File: Desktop/Domain/AlertMsgs/ConnectAlert.cs
using System;
using System.Runtime.InteropServices;
using Blurts.Domain;

namespace Blurts
{
  [ComVisible(true)]
  public class ConnectAlert : AlertBase
  {
    // Tag Names
    private static String ALERT_NAME = "Connect";
    private static String KEY_STATUS = "Text";
    private static String KEY_BT_NAME  = "BluetoothName";
    private static String KEY_DEVICE_PIN   = "DevicePIN";
    private static String KEY_DEVICE_MODEL = "DeviceModel";

    public ConnectAlert(String s)
      : base(s)
    {
    }

    public ConnectAlert()
      : base(DisplayItemType.CONNECT, ALERT_NAME)
    {
    }

    [ComVisible(true)]
    public String Text
    {
      get
      {
        return optString(KEY_STATUS);
      }
      set
      {
        try
        {
          remove(KEY_STATUS);
          put(KEY_STATUS, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String BluetoothName
    {
      get
      {
        return optString(KEY_BT_NAME);
      }
      set
      {
        try
        {
          remove(KEY_BT_NAME);
          put(KEY_BT_NAME, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String DevicePIN
    {
      get
      {
        return optString(KEY_DEVICE_PIN);
      }
      set
      {
        try
        {
          remove(KEY_DEVICE_PIN);
          put(KEY_DEVICE_PIN, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String DeviceModel
    {
      get
      {
        return optString(KEY_DEVICE_MODEL);
      }
      set
      {
        try
        {
          remove(KEY_DEVICE_MODEL);
          put(KEY_DEVICE_MODEL, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    public override Boolean ProcessMessage()
    {
      ApplicationSettings.Instance.Active = this.Activated;
      ApplicationSettings.Instance.Channel = this.Channel;
      checkVersion(this.getVersion());
      runScript( "OnConnectAlert" );
      AlertHistory.Instance.AddAlert(XML);
      return true;
    }
  }
}

File: BlackBerry/AlertMsgs/ConnectAlert.java
package com.mlhsoftware.Blurts;

import org.json.me.*;


class ConnectAlert extends AlertBase 
{
  // Tag Names
  private static final String ALERT_NAME = "Connect";
  private static final String KEY_TEXT = "Text";
  private static final String KEY_BT_NAME  = "BluetoothName";
  private static final String KEY_DEVICE_PIN   = "DevicePIN";
  private static final String KEY_DEVICE_MODEL = "DeviceModel";

  public ConnectAlert( String string ) throws JSONException
  {
    super( string );
  }

  public ConnectAlert()
  {
    super( TYPE_CONNECT, ALERT_NAME );
  }

  public void setText( String value )
  {
    try
    {
      put( KEY_TEXT, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  public void setBTName( String value )
  {
    try
    {
      put( KEY_BT_NAME, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  public void setDevicePIN( String value )
  {
    try
    {
      put( KEY_DEVICE_PIN, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  public void setDeviceModel( String value )
  {
    try
    {
      put( KEY_DEVICE_MODEL, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }
}
